packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -464,14 +464,20 @@ export default function useAutocomplete(props) {
     },
   );
 
+  const isSameValue = (value1, value2) => {
+    const label1 = value1 === null ? '' : getOptionLabel(value1);
+    const label2 = value2 === null ? '' : getOptionLabel(value2);
```

Comment:
Let's not check with strict equality, in case `value` is `undefined` or empty string `''` using a controlled Autocomplete. 
```suggestion
    const label1 = !value1 ? '' : getOptionLabel(value1);
    const label2 = !value2 ? '' : getOptionLabel(value2);
```

Reply:
In most cases, or perhaps never, value will not be false. 